test/monitoring/monitoring.go

```diff
@@ -52,34 +57,39 @@ func GetPods(kubeClientset *kubernetes.Clientset, app, namespace string) (*v1.Po
 	return pods, err
 }
 
-// Cleanup will clean the background process used for port forwarding
-func Cleanup(pid int) error {
-	ps := os.Process{Pid: pid}
-	return ps.Kill()
-}
-
 // PortForward sets up local port forward to the pod specified by the "app" label in the given namespace
-func PortForward(logf logging.FormatLogger, podList *v1.PodList, localPort, remotePort int, namespace string) (int, error) {
-	podName := podList.Items[0].Name
-	portFwdCmd := fmt.Sprintf("kubectl port-forward %s %d:%d -n %s", podName, localPort, remotePort, namespace)
-	portFwdProcess, err := executeCmdBackground(logf, portFwdCmd)
-
-	if err != nil {
-		return 0, fmt.Errorf("failed to port forward: %w", err)
+// To close the port forwarding, just close the channel
+func PortForward(logf logging.FormatLogger, config *rest.Config, clientSet *kubernetes.Clientset, pod *v1.Pod, localPort, remotePort int) (chan struct{}, error) {
+	req := clientSet.RESTClient().Post().Resource("pods").Namespace(pod.Namespace).Name(pod.Name).SubResource("portforward")
+	portForwardUrl := req.URL()
+	if !strings.HasPrefix(portForwardUrl.Path, "/api/v1") {
+		portForwardUrl.Path = "/api/v1" + portForwardUrl.Path
 	}
 
-	logf("running %s port-forward in background, pid = %d", podName, portFwdProcess.Pid)
-	return portFwdProcess.Pid, nil
-}
+	stopChan := make(chan struct{})
+	readyChan := make(chan struct{})
 
-// RunBackground starts a background process and returns the Process if succeed
-func executeCmdBackground(logf logging.FormatLogger, format string, args ...interface{}) (*os.Process, error) {
-	cmd := fmt.Sprintf(format, args...)
-	logf("Executing command: %s", cmd)
-	parts := strings.Split(cmd, " ")
-	c := exec.Command(parts[0], parts[1:]...) // #nosec
-	if err := c.Start(); err != nil {
-		return nil, fmt.Errorf("%s command failed: %w", cmd, err)
+	transport, upgrader, err := spdy.RoundTripperFor(config)
```

Comment:
Wow, this really is SPDY: https://github.com/kubernetes/kubernetes/issues/89163

Reply:
yeah :) i took that code straight from kubectl